dataplane/standalone/pkthandler/pktiohandler/pktiohandler.go

```diff
@@ -0,0 +1,244 @@
+	if err := c.Send(&fwdpb.HostPortControlRequest{Msg: &fwdpb.HostPortControlRequest_ContextId{ContextId: &fwdpb.ContextId{Id: contextID}}}); err != nil {
+		return err
+	}
+	for {
+		resp, err := c.Recv()
+		if err != nil {
+			return err
+		}
+		if resp.Create {
+			st := &status.Status{
+				Code: int32(codes.OK),
+			}
+			if err := m.createPort(resp); err != nil {
+				st = &status.Status{
+					Code:    int32(codes.Internal),
+					Message: err.Error(),
+				}
+			}
+			sendErr := c.Send(&fwdpb.HostPortControlRequest{Msg: &fwdpb.HostPortControlRequest_Status{
+				Status: st,
+			}})
+			if sendErr != nil {
+				return sendErr
+			}
+		} else {
+			p, ok := m.hostifs[resp.GetPortId()]
+			if !ok {
+				sendErr := c.Send(&fwdpb.HostPortControlRequest{Msg: &fwdpb.HostPortControlRequest_Status{
+					Status: &status.Status{
+						Code:    int32(codes.FailedPrecondition),
+						Message: fmt.Sprintf("port %v doesn't exist", resp.GetPort().GetPortId().GetObjectId().GetId()),
+					},
+				}})
+				if sendErr != nil {
+					return sendErr
+				}
+				continue
+			}
+
+			m.hostifs[resp.GetPortId()].cancelFn()
+
+			if err := p.Delete(); err != nil {
+				sendErr := c.Send(&fwdpb.HostPortControlRequest{Msg: &fwdpb.HostPortControlRequest_Status{
+					Status: &status.Status{
+						Code:    int32(codes.Internal),
+						Message: err.Error(),
+					},
+				}})
+				if sendErr != nil {
+					return sendErr
+				}
+			}
+
+			delete(m.hostifs, resp.GetPortId())
+			sendErr := c.Send(&fwdpb.HostPortControlRequest{Msg: &fwdpb.HostPortControlRequest_Status{
+				Status: &status.Status{
+					Code: int32(codes.OK),
+				},
+			}})
+			if sendErr != nil {
+				return sendErr
+			}
+		}
+	}
+}
+
+var createTAPFunc = kernel.NewTap
+
+func (m *PacketIOMgr) createPort(msg *fwdpb.HostPortControlMessage) error {
+	var p portIO
+	switch msg.GetPort().GetPortType() {
+	case fwdpb.PortType_PORT_TYPE_GENETLINK:
+		portDesc := msg.GetPort().GetGenetlink()
+		var err error
+		p, err = kernel.NewGenetlinkPort(portDesc.FamilyName, portDesc.GroupName)
+		if err != nil {
+			return err
+		}
+		log.Infof("add to new genetlink port: %v %v", portDesc.FamilyName, portDesc.GroupName)
+	case fwdpb.PortType_PORT_TYPE_TAP:
+		name := msg.GetPort().GetTap().GetDeviceName()
+		var err error
+		kp, err := createTAPFunc(name)
+		if err != nil {
+			return err
+		}
+		p = kp
+		m.dplanePortIfIndex[msg.GetDataplanePort().GetObjectId().GetId()] = kp.IfIndex()
+		log.Infof("add to new netdev port: %v", name)
+	default:
+		return fmt.Errorf("unsupported port type: %q", msg.GetPort().GetPortType())
+	}
+
+	doneCh := make(chan struct{})
+
+	m.hostifs[msg.GetPortId()] = &port{
+		portIO:   p,
+		cancelFn: func() { close(doneCh) },
+	}
+
+	m.queueRead(msg.GetPortId(), doneCh)
+
+	return nil
+}
+
+func (m *PacketIOMgr) queueRead(id uint64, done chan struct{}) {
+	p := m.hostifs[id]
+	go func() {
+		buf := make([]byte, 9100)
```

Comment:
Is this number based on anything? e.g., jumbo packet?

Reply:
added a todo for this. this is MTU commonly set in the sai calls received